### PR TITLE
fix scroll behaviour when clicking on anchors

### PR DESCRIPTION
**Pull request by fazouane-marouane**
Hi,

We're having issues with internal links generated from markdown documents. Links to the same documents (normal html anchors) did not work as expected: when clicking on the links, no scrolling was done.

This was mainly due to a "event.preventDefault()". That's what this PR is all about.

### CODE REVIEW

**Comment by MoOx** [comment]
Sorry for the delay!

Shouldn't we check that it's the same page? What happen if a click on a link with an anchor to another page?

**Comment by MoOx** [comment]
@fazouane-marouane up :)

**Comment by fazouane-marouane** [comment]
Hi

I can try checking against the current location first. But the scroll probably won’t happen in the target page.
Sorry  I’ve got nothing better to propose.
If linking to outside pages with anchors is something that’s done usually, then maybe we need a more advanced Link component.
What are your thoughts on this?

**Comment by fazouane-marouane** [comment]
@MoOx The proposed fix should be Okay now. I changed the Link component and I based it on `react-router-hash-link`.
Please, tell me what changes should I make to the PR mergeable.

File: packages/plugin-renderer-react/package.json [hunk 1]
```diff
@@ -31,6 +31,9 @@
     "isomorphic-fetch": "^2.2.1",
     "prop-types": "^15.5.8",
     "react-hot-loader": "^3.0.0-beta.7",
+    "react-router-dom": "^4.2.2",
```
Comment (MoOx) [inline packages/plugin-renderer-react/package.json]: This is not an option for us. We can't use RR v4, see https://github.com/phenomic/phenomic/issues/1130

Comment (fazouane-marouane) [reply packages/plugin-renderer-react/package.json]: Well, finally someone came up with a solution not on phenomic side. you can find it here https://github.com/parisjs/parisjs-website/commit/a044ffb5250bc6f897df7fe4a8fdb54527315ed7

It consists of having a `hashLinkScroll` function:
```js
function hashLinkScroll() {
   const { hash } = window.location
 
   if (hash !== '') {
     setTimeout(() => {
       const id = hash.replace('#', '')
       const element = document.getElementById(id)
       if (element) element.scrollIntoView()
     }, 0)
   }
 }
```

Then, you'd use it like so `<Router ... onUpdate={ hashLinkScroll } > ...</Router>`.
So maybe, we just need to document this or find a way to automatically inject the `hashLinkScroll` in phenomic once and for all. What are your thoughts on this?

**Comment by MoOx** [comment]
I guess documentation about his could be great. I will start to document the project in the coming weeks and will include this when I will add support for this. Thanks for digging into this problem!

**Pull request closed without merging**